oktaPush.go

```diff
@@ -84,8 +84,50 @@ func main() {
 	if err != nil {
 		panic(err)
 	}
+   
+    err2 := writeCredToFile(writeCredToFileInput{
+            location: "Users/mitchellchang/.aws/credentials",
+            towrite: towrite,   
+    })
     return 
 }
+
+type writeCredToFileInput struct {
+    location string
+    towrite []byte
+}
+
+func writeCredToFile(i writeCredToFileInput) error {
+	o, err := oktalib.New(&oktalib.NewInput{
+		Org:                 "dev-815627",
+		IdentityProviderArn: "arn:aws:iam::216228501626:saml-provider/Okta_2",
+		SamlURI:             "/app/amazon_aws/exkawa67iQIlhKIxE4x6/sso/saml",
+	})
+	if err != nil {
+		panic(err)
+	}
+	out, err := o.GetAwsCredentials(oktalib.GetAwsCredentialsInput{
+		RoleArn:    "arn:aws:iam::216228501626:role/devops-admin-role",
+		Expiration: 28800,
+	})
+	if err != nil {
+		panic(err)
+	}
+    deflt := []byte("[default] \n")
+    id := []byte("aws_access_key_id = " + out.AwsAccessKeyId + " \n")
+	key := []byte("aws_secret_access_key = " + out.AwsSecretAccessKey + " \n")
+	token := []byte("aws_session_token = " + out.AwsSessionToken + " \n")
+    deflt = append(deflt, id...)
+    deflt = append(deflt, key...)
+    deflt = append(deflt, token...)
```

Comment:
deflt is suppose to be `i.toWrite`. You define it outside of this function (probably in main).